platform/keysystem/worldconfig/spiresetup.go

```diff
@@ -33,5 +46,25 @@ func LoadSpireSetup(path string) (*SpireSetup, error) {
 	if err != nil {
 		return nil, err
 	}
+	// validation steps
+	for _, node := range setup.Nodes {
+		if node.Kind != "worker" && node.Kind != "supervisor" && node.Kind != "master" {
+			return nil, fmt.Errorf("unrecognized kind of node: %s", node.Kind)
+		}
+		node.netIP = net.ParseIP(node.IP)
+		if node.netIP == nil {
+			return nil, fmt.Errorf("could not parse IP: %s", node.IP)
+		}
+	}
+	dupcheck := map[string]bool{}
```

Comment:
nit: sets should be maps to unit type, not maps to bool.

Reply:
I think the code here is slightly more confusing like that, but I don't care enough, so I'll make the change.

Reply:
I think it is clearer given that a map to unit is recognized as idiomatic for a set.

Reply:
Though really I don't have a strong preference.  If you like it the other way feel free to leave it.